lib/routes/routeBackbeat.js

```diff
@@ -464,6 +464,47 @@ function putData(request, response, bucketInfo, objMd, log, callback) {
     });
 }
 
+/**
+ * @callback CanonicalIdCallback
+ * @param {Error|null} err - Error object if operation failed, null otherwise
+ * @param {Object} [result] - Result object containing account information
+ * @param {string} [result.accountId] - The account ID
+ * @param {string} [result.canonicalId] - The canonical ID associated with the account
+ * @param {string} [result.name] - The display name associated with the account
+ * @returns {undefined}
+ */
+
+/**
+ * 
+ * @param {string} accountId - account ID
+ * @param {Log} log - logger instance
+ * @param {CanonicalIdCallback} cb - callback function
+ * @returns {undefined}
+ */
+function getCanonicalIdsByAccountId(accountId, log, cb) {
+    if (!accountId) {
+        return cb();
+    }
```

Comment:
Should we add a debug log here?

Reply:
should be moved out of this function: for readability, best to keep the logic of "skipping" the call at the caller's.... where it is also easier to answer our question (_is a log needed?_)
→ by moving this condition to `putMetadata()`, it will be clear we only do it if the account is provided (possibly with a comment explaining in which case it happens)

(if we really need to keep the test here, then we should try and make it clear the function does not always get the canonicalId : e.g. something like `maybeGetCanonicalIdsByAccountId`. At leat it gives an indication to the reader that it may not be done, but it still forces to look through the abstraction, and hence IMHO not as readable...)